numba_dpex/experimental/_kernel_dpcpp_spirv_overloads/_item_overloads.py

```diff
@@ -0,0 +1,96 @@
+# SPDX-FileCopyrightText: 2023 - 2024 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Implements the SPIR-V overloads for the kernel_api.items class methods.
+"""
+
+from numba.core import types
+from numba.core.errors import TypingError
+from numba.extending import intrinsic, overload_method
+
+from numba_dpex.experimental.core.types.kernel_api.items import (
+    ItemType,
+    NdItemType,
+)
+from numba_dpex.ocl._declare_function import _declare_function
+
+from ..target import DPEX_KERNEL_EXP_TARGET_NAME
+
+
+@intrinsic(target=DPEX_KERNEL_EXP_TARGET_NAME)
+def _intrinsic_get_global_id(
+    ty_context, ty_dim  # pylint: disable=unused-argument
+):
+    """Generates instruction for spirv i64 get_global_id(i32) call."""
+    sig = types.int64(types.int32)
+
+    def _intrinsic_exchange_gen(context, builder, sig, args):
+        [dim] = args
+        get_global_id = _declare_function(
+            # unsigned int - is what demangler returns from IR instruction
+            # generated by dpcpp. However int32 is passed as an argument.
```

Comment:
Actually, I think it is the reason Numba generates a lot of extra bound checks as we are passing a signed int to a function that expects unsigned int. See https://github.com/IntelPython/numba-dpex/issues/936. Filing it as a future TODO unrelated to current PR.